src/open-uri.c

```diff
@@ -107,19 +108,26 @@ get_latest_choice_info (const char *app_id,
               permissions != NULL &&
               permissions[0] != NULL)
             {
-              g_auto(GStrv) permission_detail = g_strsplit (permissions[0], ":", 2);
-              if (g_strv_length (permission_detail) >= 2)
+              guint n_permissions;
+
+              g_auto(GStrv) permission_detail = g_strsplit (permissions[0], ":", 3);
+              n_permissions = g_strv_length (permission_detail);
```

Comment:
I think all of this parsing is a bit annoying; it would be nicer to use the structure of the database instead of stuffing everything into one string. That could be done in several ways:
- just use 3 separate strings, instead of splitting one at ":"
- maybe use a separate table row for the thresholds, to cleanly separate them from the usage data